gossip/comm/comm_test.go

```diff
@@ -624,35 +624,56 @@ func TestCommSend(t *testing.T) {
 		}
 	}()
 
-	c1recieved := 0
-	c2recieved := 0
+	c1received := 0
+	c2received := 0
 	// hopefully in some runs we'll fill both send and receive buffers and
 	// drop overflowing messages, but still finish, because the endless
 	// stream of messages inexorably gets through unless something is very
 	// broken.
-	totalMessagesRecieved := (DefSendBuffSize + DefRecvBuffSize) * 2
-RECV:
-	for {
-		select {
-		case <-ch1:
-			c1recieved++
-			if c1recieved == totalMessagesRecieved {
-				close(stopch2)
-			}
-		case <-ch2:
-			c2recieved++
-			if c2recieved == totalMessagesRecieved {
-				close(stopch1)
+	totalMessagesReceived := (DefSendBuffSize + DefRecvBuffSize) * 2
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	// read from each until there are an acceptable number of
+	// messages and then stop the sender from sending more
+	// messages.
+	go func() {
```

Comment:
1) Why not close stopCh1 in a defer when this go func exits?
2) Can't we create a function that spawns this go func and then just invoke it 2 times for ch1 and ch2 ? (to reduce code duplication)